src/script/printer/utils.h

```diff
@@ -161,13 +161,25 @@ inline Optional<String> FindFunctionName(const IRDocsifier& d, const BaseFunc& f
 }
 
 inline String GenerateUniqueName(std::string name_hint,
-                                 const std::unordered_set<String>& defined_names) {
+                                 const std::unordered_set<String>& defined_names,
+                                 const Array<String>& binding_names) {
   for (char& c : name_hint) {
     if (c != '_' && !std::isalnum(c)) {
       c = '_';
     }
   }
   std::string name = name_hint;
+  // if the name matches the name currently being bound, then do not add a suffix
+  // (this comes up in the case of defining a local function: the local function
+  // is the RHS of a binding. The var name on the LHS will have been looked at first;
+  // without this check, we would print a suffix on the function name even though it
+  // is actually the first definition)
+  for (const auto& bound_name : binding_names) {
+    if (name == bound_name) {
+      return name;
+    }
+  }
+
```

Comment:
Fix to the bug noted in `test_tvmscript_printer_relax.py`, except I'm not sure if this is the best approach to detecting the case of a function definition. This is a result of the handling of `VarBinding`s in `src/script/printer/relax/binding.cc`

Reply:
I think this could have false positives, like if you use the name of the function currently being bound inside the function's body, so I might have to think of a better way to handle the case of a nested function.

Reply:
Found a better alternative solution